app/src/components/CustomPeriodDialog.tsx

```diff
@@ -25,16 +25,20 @@ export function CustomPeriodDialog(props: CustomPeriodDialogProps) {
   const [endTime, setEndTime] = useState<TimeValue>(new Time(12, 0));
 
   const [startDate, setStartDate] = useState<DateValue>(
-    toCalendarDate(new Date(Date.now() - PeriodProps[Period.WEEK].milliseconds))
+      toCalendarDate(new Date(
+          Math.max(Date.now() -
+              PeriodProps[Period.WEEK].milliseconds,
+              new Date('2013-01-01T00:00:00').getTime())))
   );
 
   const [endDate, setEndDate] = useState<DateValue>(toCalendarDate(new Date()));
 
   function handleSelection() {
-    const startDateTime = toDate(startDate, startTime);
+    const startDateTime = Math.max(toDate(startDate, startTime).getTime(),
+        new Date('2013-01-01T00:00:00').getTime());
     const endDateTime = toDate(endDate, endTime);
 
-    onSelected(startDateTime, endDateTime);
+    onSelected(new Date(startDateTime), new Date(endDateTime));
```

Comment:
With your changes `startDateTime` now has the type of `number`, but if you wrap it in a Date object, you don't need to do it down here, so:

`const startDateTime = new Date(Math.max(toDate(startDate, startTime).getTime(), MIN_DATE.getTime()));`
...
`onSelected(startDateTime, endDateTime);`